src/ui/Deck.js

```diff
@@ -1,11 +1,26 @@
 import Api from "./Api.js";
+import Hand from "./Hand.js";
+import { useState, useCallback } from "react";
 
-class Deck(){
-	constructor(deckCode){
-	
-	 static async newDeck = await().Api.Deck(deckCode);
-	 this.Deck = newDeck
+export default class Deck {
```

Comment:
This is where it would be good to have an async factory function, something like this maybe?

```
static async create() {
  const id = await Api.deck(); // create the deck on the server and get the id
  return new Deck(id);
}
```
Disclaimer: I have not actually run this code :P